packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -913,7 +914,7 @@ class NavigatorState extends State<Navigator> with TickerProviderStateMixin {
   ///
   /// Returns a [Future] that completes to the `result` value passed to [pop]
   /// when the pushed route is popped off the navigator.
-  Future<dynamic> push(Route<dynamic> route) {
+  Future<Object> push(Route<Object> route) {
```

Comment:
I had originally made this generic as well, but "--no-implicit-dynamic" doesn't know about @optionalTypeArgs, and so some explicit `push<dynamic>` calls were required.  Using `Object` here avoids introducing `dynamic` at all.